docs/source/data/resolvers.mdx

```diff
@@ -388,49 +374,75 @@ Resolver functions are passed four arguments: `parent`, `args`, `context`, and `
+To provide an initial `context` to your resolvers, add a `context` initialization function to your integration function (e.g., `expressMiddleware` or `startStandaloneServer`). This function is called _with every request_, so you can customize the context based on each request's details (such as HTTP headers).
 
-```js
+<Disclaimer />
 
-// Constructor
-const server = new ApolloServer({
+<MultiCodeBlock>
+
+```ts
+import { GraphQLError } from 'graphql';
+
+interface MyContext { // Context typing
+  authScope?: String;
+}
+
+const server = new ApolloServer<MyContext>({
   typeDefs,
   resolvers,
-  csrfPrevention: true,
-  cache: 'bounded',
-  context: ({ req }) => ({
-    authScope: getScope(req.headers.authorization)
-  }),
-  plugins: [
-    ApolloServerPluginLandingPageLocalDefault({ embed: true }),
-  ],
 });
 
+const { url } = await startStandaloneServer(server, {
+  // A named context function is required if you are not
+  // using ApolloServer<BaseContext>
+  context: async ({req, res}) => ({
+      authScope: getScope(req.headers.authorization),
+  }),
+  listen: { port: 4000 }
+})
+
 // Example resolver
 (parent, args, context, info) => {
-  if(context.authScope !== ADMIN) throw new AuthenticationError('not admin');
+  if (context.authScope !== ADMIN) throw new GraphQLError("not admin!", {
+  extensions: {
+    code: 'UNAUTHENTICATED',
+  },
+});
   // Proceed
-}
+};
 ```
 
-> This example assumes you're using either the `apollo-server` or `apollo-server-express` package, both of which use Express. The fields of the object passed to your `context` function might differ if you're using middleware besides Express. [See the API reference for details.](../api/apollo-server/#middleware-specific-context-fields)
->
->For more information on middleware in general, see [Choosing an Apollo Server package](../integrations/middleware/).
+</MultiCodeBlock>
+
+> This example assumes you're using either the `startStandaloneServer` or `expressMiddleware`, both of which use Express under the hood. The fields of the object passed to your `context` function might differ if you're using a different integration.
+
+The `context` function should return an object, which is then accessible to your server's resolvers and plugins (via the `contextValue` field).
 
-Context initialization can be asynchronous, allowing database connections and other operations to complete:
+<!-- TODO (AS4): update these links once we have an article with other integration functions
+
+[For more details on Apollo Server integrations, see [Choosing an Apollo Server package](/apollo-server/api/apollo-server/#middleware-specific-context-fields).
+-->
+
+Context initialization can be asynchronous, enabling database connections and other operations to complete:
```

Comment:
In fact, in AS4, the TS types *require* it to be an async function.

Reply:
Added a mention to this effect above and changed this to:

> Because the `context` initialization function is asynchronous, you can use it to establish database connections and wait for other operations to complete: